src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SymbolToProjectId.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial class SolutionState
+    {
+        /// <summary>
+        /// Contains caches to help us quickly and accurately map from <see cref="ISymbol"/>s to <see
+        /// cref="ProjectId"/>.
+        /// </summary>
+        /// <remarks>
+        /// Note: this is not <see langword="readonly"/> as this is a struct and will mutate as operations are called on
+        /// it.  However, in the common case where this data is not needed, it will have not have any impact.
+        /// </remarks>
+        private SymbolToProjectId _symbolToProjectId;
+
+        /// <inheritdoc cref="Solution.GetExactProjectId"/>
+        public ProjectId? GetExactProjectId(ISymbol symbol)
+            => _symbolToProjectId.GetExactProjectId(symbol);
+
+        /// <summary>
+        /// A helper type that can be used to map from symbols to a project they could have come from.
+        /// </summary>
```

Comment:
Why do we have this struct? Just for scoping of fields?

Reply:
yes.

Reply:
Removed.  Now that we're down to one field, we don't need it :)